module2/src/com/epam/decomposition/NumbersAnalyzer.java

```diff
@@ -0,0 +1,247 @@
+            for (int i = 1; i < array.length; i++) {
+                if (array[i] < array[i - 1]) {
+                    int temp = array[i - 1];
+                    array[i - 1] = array[i];
+                    array[i] = temp;
+                    isSorted = false;
+                }
+            }
+        }
+        return array[array.length  - indexNumber];
+    }
+
+    //task 6
+    public static boolean checkThreeNumbersMutuallyPrime(int firstNumber, int secondNumber, int thirdNumber) {
+        if (findGreatestCommonDivisor(findGreatestCommonDivisor(firstNumber,secondNumber), thirdNumber) == 1) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    //task 7
+    public static int calculateSumFactorialsOddNumbers(int minBorder, int maxBorder) {
+        int sumFactorials = 0;
+        for (int i = minBorder; i <= maxBorder; i += 2) {
+            sumFactorials += calculateFactorialNumber(i);
+        }
+        return sumFactorials;
+    }
+
+    private static double calculateFactorialNumber(double number) {
+        for(double i = number-1; i > 1; i--) {
+            number *= i;
+        }
+        return number;
+    }
+
+    //task 8
+    public static int calculateSumThreeConsecutiveElementsOfArray(int[] array, int firstIndex, int lastIndex) {
+        int sumElements = 0;
+        for(int i = firstIndex; i <= lastIndex; i++) {
+            sumElements += array[i];
+        }
+        return sumElements;
+    }
+
+    //task 10
+    public static int[] calculateArrayFormationFromNumber(int number) {
+        int[] array = new int[findNumberLength(number)];
+        int index = array.length - 1;
+        while (number > 0) {
+            array[index--] = number % 10;
+            number /= 10;
+        }
+        return array;
+    }
+
+    private static int findNumberLength(int number) {
+        int numberLength = 0;
+        while (number > 0) {
+            numberLength++;
+            number /= 10;
+        }
+        return numberLength;
+    }
+
+    //task 11
+    public static int  calculateWhichNumberIsLonger (int firstNumber, int secondNumber) {
+        int lengthFirstNumber = findNumberLength(firstNumber);
+        int lengthSecondNumber = findNumberLength(secondNumber);
+        if (lengthFirstNumber > lengthSecondNumber) {
+            return firstNumber;
+        } else {
+            return secondNumber;
+        }
+    }
+
+    //task 13
+    public static void findPairsOfSimpleTwins (int startingPoint) {
+        for (int i = startingPoint; i <= startingPoint * 2; i++) {
+            if (checkForPrimeNumber(i) && checkForPrimeNumber(i + 2)) {
+                System.out.println(i + " " + (i + 2));
+            }
+        }
+    }
+
+    private static boolean checkForPrimeNumber (int number) {
+        for (int i = 2; i <= number / 2; i++) {
+            if (number % i == 0) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
In your implementation, you will check each number for simplicity 2 times, which is not efficient. I can show you a more competent solution:
```suggestion
    public static StringBuilder findTwinPrimeNumbers(int firstNumber, int lastNumber) {
        StringBuilder twinPrimes = new StringBuilder();
        final int specifiedFactor = 2;
        if (firstNumber > 2 && lastNumber == specifiedFactor * firstNumber) {
            ArrayList<Integer> primeNumbers = findPrimeNumbers(firstNumber, lastNumber);
            if (primeNumbers.size() != 0) {
                for (int i = 1; i < primeNumbers.size(); i++) {
                    if (primeNumbers.get(i - 1) + 2 == primeNumbers.get(i)) {
                        twinPrimes.append("[").append(primeNumbers.get(i - 1)).append(", ").append(primeNumbers.get(i)).
                                append("] ");
                    }
                }
            }
        }
        return twinPrimes;
    }

    private static ArrayList<Integer> findPrimeNumbers(int firstNumber, int lastNumber) {
        final int minDivisorForPrimeNumbers = 2;
        final int firstPrimeNumber = 2;
        ArrayList<Integer> primeNumbers = new ArrayList<>();
        if (firstNumber >= firstPrimeNumber && lastNumber > firstNumber) {
            for (int i = firstPrimeNumber; i <= lastNumber; i++) {
                boolean isCompositeNumber = false;
                for (int divisor = minDivisorForPrimeNumbers; divisor <= Math.round(Math.sqrt(i)); divisor++) {
                    if (i % divisor == 0) {
                        isCompositeNumber = true;
                        break;
                    }
                }
                if (!isCompositeNumber) {
                    primeNumbers.add(i);
                }
            }
        }
        return primeNumbers;
    }
```